### PR TITLE
Made regex configurable for identifier

### PR DESCRIPTION
**Pull request by spaiva**
This PR goes in conjunction with LORIS MRI [PR#152](https://github.com/aces/Loris-MRI/pull/152)
- Made regular expression configurable at the top of the file
- No longer using `$lookupCenterNameUsing` as main identifier variable
- Removed `getSubjectIDs()` as duplicate method in MRI.pm is more robust than the one implemented here.

### CODE REVIEW

File: profileTemplate [hunk 1]
```diff
@@ -36,18 +36,17 @@ $mail_user      = "yourname\@example.com";      # who will be notified
 $get_dicom_info = "/PATH/TO/get_dicom_info.pl"; # path to get_dicom_info.pl
 $horizontalPics = 1;
 $no_nii         = 0;                            # 0=create nifti files, 1=don't create nifti files
+$regex_pattern  = '([^_]+)_(\d+)_([^_]+)'       # regular expression to parse PatientName or PatientID
```
Comment (driusan) [inline profileTemplate]: this should probably have a more descriptive name like "$PatientName_regex"

Comment (stronnics) [inline profileTemplate]: Agreed.

File: profileTemplate [hunk 2]
```diff
@@ -88,40 +87,40 @@ sub getIncomingDir {
     my $site = shift;
     return $tarchiveLibraryDir;
 }
-# extracts the subject and timepoint identifiers from the patient name 
+# extracts the subject and timepoint identifiers from the patient name
 # assumes identifers are stored as <PSCID>_<DCCID>_<visit> in PatientName field, where <visit> is 3 digits.
 sub getSubjectIDs {
-    my ($patientName, $patientID, $scannerID, $dbhr) = @_;
+    my ($identifier, $patientID, $scannerID, $dbhr) = @_;
```
Comment (MounaSafiHarab) [inline profileTemplate]: @stronnics,
shouldn't this be updated to take in 3 arguments?

Comment (driusan) [inline profileTemplate]: wouldn't that involve also changing the Loris-MRI code?

Comment (stronnics) [inline profileTemplate]: Mouna is correct, LORIS-MRI code uses 3 arguments. I made the change.

File: profileTemplate [hunk 3]
```diff
@@ -36,18 +36,17 @@ $mail_user      = "yourname\@example.com";      # who will be notified
 $get_dicom_info = "/PATH/TO/get_dicom_info.pl"; # path to get_dicom_info.pl
 $horizontalPics = 1;
 $no_nii         = 0;                            # 0=create nifti files, 1=don't create nifti files
+$PatientID_regex  = '([^_]+)_(\d+)_([^_]+)'       # regular expression to parse PatientName or PatientID
```
Comment (cmadjar) [inline profileTemplate]: I feel the regex should be part of Settings. It could change according to the different projects

Comment (cmadjar) [inline profileTemplate]: haha. Ooops. I just realized I made a stupid comment. ;) Never mind.

Sorry about that.

File: profileTemplate [hunk 4]
```diff
@@ -29,25 +29,24 @@
...
-$horizontalPics = 1;
-$no_nii         = 0;                            # 0=create nifti files, 1=don't create nifti files
+$data_dir        = '/PATH/TO/DATA/location';     # where the data goes
+$bin_dir         = '/PATH/TO/BIN/location';      # where the imaging scripts go
+$prefix          = "project";                    # study prefix or study name
+$mail_user       = "yourname\@example.com";      # who will be notified
+$get_dicom_info  = "/PATH/TO/get_dicom_info.pl"; # path to get_dicom_info.pl
+$horizontalPics  = 1;
+$no_nii          = 0;                            # 0=create nifti files, 1=don't create nifti files
+$PatientID_regex = '([^_]+)_(\d+)_([^_]+)'       # regular expression to parse PatientName or PatientID
```
Comment (cmadjar) [inline profileTemplate]: @spaiva 
For the regex, I had to modify it to: ([^_]+)_(\d+)_([^_]+)|(lego_phantom)|(living_phantom) to allow insertion of phantom scans as well. Several projects use them and it is how they are labelled.

Other than that, all good! :) Let me know when you make the change and I'll check the PassedCodeReview.

Comment (spaiva) [inline profileTemplate]: @cmadjar I see. Has it been tested with this configuration? I will test it on my end and update the PR accordingly.

Comment (cmadjar) [inline profileTemplate]: @spaiva 

Yes, I tested a candidate and a phantom dataset on my sandbox. :) Does not hurt to test it on your end too though.

**Comment by MounaSafiHarab** [comment]
@spaiva, @cmadjar 
I will set the milestone to 17.1 to be in sync with redmine I guess.
I will remove the PassedCodeReview as the Loris-MRI one still needs work...

**Comment by spaiva** [comment]
Ready for testing again @MounaSafiHarab and @cmadjar, sorry for the long delay. Basically, it is better to use the `getSubjectIDs` method from MRI.pm than from profileTemplate. I made changes accordingly and would like to have your inputs. Thanks.

**Comment by MounaSafiHarab** [comment]
I added the needs work label just to make sure that the changes are still valid and do not conflict with other requests that have been merged, since it has been a while (I am no longer sure which release this is going to)

**Pull request closed without merging**